addons/website/models/website.py

```diff
@@ -192,6 +192,21 @@ def _compute_menu(self):
     def _get_menu_ids(self):
         return self.env['website.menu'].search([('website_id', '=', self.id)]).ids
 
+    @tools.ormcache('self.env.uid', 'self.id')
+    def has_record_url_in_menu(self):
+        """
+        Check if the website menu contains a record like url.
+
+        :return: True if the menu contains a url like /XXX/YYY-ZZZ
+        """
+        for url in self.env['website.menu'].search([('website_id', '=', self.id)]).mapped('url'):
+            if url and url.startswith('/') and url.count('/') > 1:
+                last_part = url.split('/')[-1]
+                dash_parts = last_part.split('-')
+                if len(dash_parts) > 1 and len(dash_parts[-1]) > 1 and len(dash_parts[-2]) > 1:
+                    return True
+        return False
```

Comment:
Couldn't we use the regex existing in `addons/website/tests/test_crawl.py`?
```suggestion
        return any(self.env['website.menu'].search([('website_id', '=', self.id)]).filtered(lambda menu: re.match(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", menu.url)))
```

And actually (use of `_get_menu_ids`):
```suggestion
        return any(self.env['website.menu'].browse(self._get_menu_ids()).filtered(lambda menu: re.match(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", menu.url)))
```

I am still looking for the perf query issue

Reply:
I hate regexes but yes, as the regex already exists, I will use it :+1: Thanks

Reply:
Same, but if it's commented to explain what it does it's ok, here something like `The regex is checking is there is a slug (/text-id) inside an url` I guess.
Especially when it reduce the code that much, here it's really way shorter.
What's criminal is regex without any explanation :')

Reply:
`re.match(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-world-1")`  doesn't match.. I really think it's better to keep 5 lines of understandable string manipulation rather than trying to create an incomprehensible regex to make the code more concise. IMO code readability is more important than code length.

Reply:
It works, since our crawler is using this regex to parse all our controllers.
It's just that it was `re.search` and not `re.match`.
I also despise regex in general, but in this case it's the way to go imo.. Especially since you code was not working actually, it wouldn't catch `/event/event-3/register` while the regex correctly does:
```py
re.search(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-hello-1/ok")
<re.Match object; span=(5, 20), match='/hello-hello-1/'>
```
It was also not working with `/blog/blog-2`, it needed a double digit like `/blog/blog-22`
```py
def gdi(url):
     if url and url.startswith('/') and url.count('/') > 1:
         last_part = url.split('/')[-1]
         dash_parts = last_part.split('-')
         if len(dash_parts) > 1 and len(dash_parts[-1]) > 1 and len(dash_parts[-2]) > 1:
             return True
     return False
 
gdi("/blog/hello-hello-1")
False  # wrong
gdi("/blog/hello-1")
False  # wrong
gdi("/blog/hello-11")
True # ok
gdi("/blog/hello-11/ok")
False # wrong
```

With the regex, everything is fine:
```py
re.search(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-11/ok")
<re.Match object; span=(5, 15), match='/hello-11/'>
re.search(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-11")
<re.Match object; span=(5, 14), match='/hello-11'>
re.search(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-1")
<re.Match object; span=(5, 13), match='/hello-1'>
re.search(r"[/](([^/=?&]+-)?[0-9]+)([/]|$)", "/blog/hello-hello-1")
<re.Match object; span=(5, 19), match='/hello-hello-1'>
```